mailpoet/assets/js/src/automation/components/status/index.tsx

```diff
@@ -0,0 +1,60 @@
+import classNames from 'classnames';
+import { automationRunStatusNames, automationStatusNames } from './names';
+import {
+  automationRunStatusClasses,
+  automationStatusClasses,
+  orderStatusClasses,
+} from './classes';
+
+type StatusBadgeProps = {
+  name: string;
+  className: string;
+};
+
+type StatusProps = {
+  status: string;
+};
+
+type OrderStatusProps = StatusProps & {
+  name: string;
+};
+
+export function StatusBadge({
+  name,
+  className,
+}: StatusBadgeProps): JSX.Element {
+  return (
+    <span
+      className={classNames(
+        'mailpoet-automation-status',
+        `mailpoet-automation-status-${className}`,
+      )}
+    >
+      {name}
+    </span>
+  );
+}
+
+export function AutomationStatus({ status }: StatusProps): JSX.Element {
+  return (
+    <StatusBadge
+      name={automationStatusNames[status] ?? status}
+      className={automationStatusClasses[status] ?? status}
+    />
+  );
+}
+
+export function AutomationRunStatus({ status }: StatusProps): JSX.Element {
+  return (
+    <StatusBadge
+      name={automationRunStatusNames[status] ?? status}
+      className={automationRunStatusClasses[status] ?? status}
+    />
+  );
+}
+
+export function OrderStatus({ status, name }: OrderStatusProps): JSX.Element {
+  return (
+    <StatusBadge name={name} className={orderStatusClasses[status] ?? status} />
+  );
+}
```

Comment:
Technically, this one should probably go to the MailPoet integration where it is used (an order is not something known to the automation engine), but it's not a very important thing, we're not always 100% strict with the separation of these. I'll leave the decision up to you.

Reply:
@JanJakes you meant something like this? https://github.com/mailpoet/mailpoet/pull/5696/commits/b95ff401fb8db8410e73535ec8ab0cb7c2304a98